### DISCUSSION TITLE
Restarting notebook from within the notebook

### DISCUSSION THREAD

**Discussion by mossr** [fonsp/Pluto.jl#2205] [Pluto development]
Would it be possible to add a button inside the notebook that performs a notebook restart like the current process below (somewhat tedious when restarting a bunch 🙃)
![image](https://user-images.githubusercontent.com/55111154/100132272-2df13580-2eab-11eb-8bbe-4ca1fb630a0e.gif)

Btw, I couldn't find other discussion about this type of feature, and I'm also not sure if this is the best place for this discussion.

**Comment by rikhuijzer** [comment]
I agree. Related discussion at https://discourse.julialang.org/t/adding-a-restart-process-button-in-pluto/76812 (without a solution)

**Reply by mossr** [reply]
That's an excellent option in the meantime, great work

**Reply by mossr** [reply]
Thank you!

**Comment by disberd** [comment]
It seem you can now do a restart with the following cell code:
```
html"""
<script>
	const cell = currentScript.closest('pluto-cell')
	cell._internal_pluto_actions.send(                    
		    "restart_process",
                    {},
                    {
                        notebook_id: editor_state.notebook.notebook_id,
                    }
                )
</script>
"""
```
Probably at the time @rikhuijzer asked the question earlier this year the `pluto_actions` were not exposed as part of the cell objects.

If you want a button (also because just pasting the above in a cell will simply start an infinite restart loop as I just realized :D), you can adapt @rikhuijzer's last code snippet from discourse as such:
```
HTML("""
<!-- the wrapper span -->
<div>
	<button id="myrestart" href="#">Restart</button>
	
	<script>
		const div = currentScript.parentElement
		const button = div.querySelector("button#myrestart")
		const cell= div.closest('pluto-cell')
		console.log(button);
		button.onclick = function() { restart_nb() };
		function restart_nb() {
			console.log("Restarting Notebook");
		        cell._internal_pluto_actions.send(                    
		            "restart_process",
                            {},
                            {
                                notebook_id: editor_state.notebook.notebook_id,
                            }
                        )
		};
	</script>
</div>
""")
```
